src/main/java/io/stargate/sgv2/jsonapi/service/operation/tables/WhereCQLClauseAnalyzer.java

```diff
@@ -239,6 +241,39 @@ private void checkFilteringOnComplexColumns(Map<CqlIdentifier, TableFilter> iden
     }
   }
 
+  /**
+   * Check if any non $eq filter is used.
+   *
+   * <p>e.g. Table UpdateOne and DeleteOne, DATA API does not know the exact modified row count in
+   * DB, so it needs to restrict on the filters. Make sure only $eq is used for DeleteOne and
+   * UpdateOne.
+   */
+  private void checkOnlyEqUsage(Map<CqlIdentifier, TableFilter> identifierToFilter) {
+
+    var nonEqFilterColumns =
+        identifierToFilter.entrySet().stream()
+            .filter(
+                entry -> {
+                  TableFilter tableFilter = entry.getValue();
+                  if (!(tableFilter instanceof NativeTypeTableFilter<?> nativeTypeTableFilter)) {
```

Comment:
Why is this needed? And should it really return `true` (included as failing filter) instead of `false` (ignored, I think)

Reply:
only nativeTypeTableFilter has $eq, so if a tableFilter is not a nativeTypeTableFilter, then it is using operator other than $eq, so return true to keep that in the filter stream.